openedx_external_enrollments/external_enrollments/salesforce_external_enrollment.py

```diff
@@ -192,34 +188,65 @@ def _get_program_of_interest_data(self, data, order_lines):
             )
             program_of_interest["Secondary_Source"] = program_of_interest.get(
                 "Secondary_Source",
-                ""
+                "",
             )
         except Exception:  # pylint: disable=broad-except
             pass
 
         return program_of_interest
 
+    def _get_program_course_runs(self, data):
+        """Returns the list with the course runs of the courses associated with the program."""
+        course_runs = []
+        program = data.get("program")
+        if program:
+            course_runs = [
+                run["key"] for course in program.get("courses", []) for run in course.get("course_runs", [])
+            ]
+
+        return course_runs
+
     def _get_courses_data(self, data, order_lines):  # pylint: disable=unused-argument
         """
+        Retrieves the following data per course:
+        - CourseName
+        - CourseID
+        - CourseRunID
+        - CourseStartDate
+        - CourseEndDate
+        - CourseDuration
+        - Institution_Hidden
+        - Program_of_Interest
+
+        Institution_Hidden and Program_of_Interest depends on whether the course is
+        part of a program or is just a single course.
 
         :param data:
         :param order_lines:
-        :return:
+        :returns: courses (List of dicts):
         """
         courses = []
         for line in order_lines:
             try:
                 course_id = line.get("course_id")
                 course = self._get_course(course_id)
                 course_key = self._get_course_key(course_id)
-                salesforce_settings = course.other_course_settings.get("salesforce_data")
+                salesforce_settings = course.other_course_settings.get("salesforce_data", {})
                 course_data = dict()
                 course_data["CourseName"] = salesforce_settings.get("Program_Name") or course.display_name
                 course_data["CourseID"] = "{}+{}".format(course_key.org, course_key.course)
                 course_data["CourseRunID"] = course_id
                 course_data["CourseStartDate"] = self._get_course_start_date(course, line.get("user_email"), course_id)
                 course_data["CourseEndDate"] = course.end.strftime("%Y-%m-%d")
                 course_data["CourseDuration"] = "0"
+                course_data["Institution_Hidden"] = salesforce_settings.get("Institution_Hidden")
+                course_data["Program_of_Interest"] = salesforce_settings.get("Program_of_Interest")
```

Comment:
Why don't you preserve the camel case style on course_data keys?

```suggestion
                course_data["ProgramOfInterest"] = salesforce_settings.get("Program_of_Interest")
```

Reply:
@ivanvgh That's how is specified. I know it seems weird. If you see the ticket you will find out that camel case and snake case are present. However thanks for the observation, I will ask.